e2ev2/tests/workspace-create.test.js

```diff
@@ -0,0 +1,70 @@
+  await page.waitForSelector('[data-test-id="billing-dropdown-div"]')
+  const createButton = await page.waitForFunction(
+    () => [...document.querySelectorAll('[role="button"]')]
+      .filter(n => n.innerText.toLowerCase() === 'create workspace')[0])
+  expect(await createButton.evaluate(n => n.style.cursor)).toBe('not-allowed')
+  const wsName = `test-ws-share-${utils.denseDateTime()}`
+  await page.type('[placeholder="Workspace Name"]', wsName)
+  await page.type('#education-purpose', ' ') // ???!!!
+  await page.type('#intendedStudyText', 'Does this work?')
+  await page.type('#scientificApproachText', 'Automated regression testing.')
+  await page.type('#anticipatedFindingsText', 'What, if anything, is broken.')
+  expect(await page.waitForSelector('[data-test-id="otherDisseminateResearch-text"]')
+    .then(eh => eh.evaluate(e => e.disabled)))
+    .toBe(true)
+  // Why doesn't this work?
+  // await page.click('input[data-test-id="OTHER-checkbox"]')
+  await page.waitForSelector('input[data-test-id="OTHER-checkbox"]')
+    .then(eh => eh.evaluate(e => e.click()))
+  expect(await page.waitForSelector('[data-test-id="otherDisseminateResearch-text"]')
+    .then(eh => eh.evaluate(e => e.disabled)))
+    .toBe(false)
+  await page.type('textarea[data-test-id="otherDisseminateResearch-text"]', 'Team test reports.')
+  await page.evaluate(() =>
+    Array.from(document.querySelectorAll('input[type="checkbox"]'))[17].click()) // no
+  await page.click('input[type="radio"][data-test-id="specific-population-no"]')
+  await page.click('div > div + div > input[type="radio"][name="reviewRequested"]')
+  expect(await createButton.evaluate(n => n.style.cursor)).toBe('pointer')
+  await createButton.click()
+  await page.click('[role="dialog"] [role="button"] + [role="button"]')
+  // Allow 30 seconds for workspace creation.
+  const workspacePageTitleLink =
+    await page.waitForSelector('a[href="/workspaces"] + span + div > a',
+    {timeout: workspaceCreationTimeoutMs})
+  expect(await workspacePageTitleLink.evaluate(e => e.innerText)).toBe(wsName)
+  await page.click('[data-test-id="workspace-menu-button"]')
+  // side menu pops up
+  await page.click('div#popup-root clr-icon[shape="trash"]')
+  // modal confirmation pops up
+  const confirmButton =
+    await page.waitForSelector('[role="dialog"] [role="button"] + [role="button"]')
+  expect(await confirmButton.evaluate(n => n.style.cursor)).toBe('not-allowed')
+  await page.type('[role="dialog"] input[placeholder="type DELETE to confirm"]', 'delete')
+  expect(await confirmButton.evaluate(n => n.style.cursor)).toBe('pointer')
+  await confirmButton.click()
+  // automatic navigation to Workspaces page
+  expect(await page.waitForSelector(
+    '[data-test-id="signed-in"] > div:nth-child(2) h3', {timeout: workspaceDeletionTimeoutMs}
+    ).then(eh => eh.evaluate(e => e.innerText)))
+    .toBe('Workspaces')
+}, workspaceCreationTimeoutMs + workspaceDeletionTimeoutMs + 10e3)
```

Comment:
What does `10e3` represent?

Reply:
Providing a number here overrides the default test timeout (5 seconds) for this particular test. It is in milliseconds. `10e3` is just a little easier for me to read than `10000`, as the zeros get lost for me.

Reply:
It looks like the timeout is being overridden with workspaceCreationTimeoutMs (30 seconds) + workspaceDeletionTimeoutMs (10 seconds) + 10e3 (an additional 10 seconds). The first two variables explain to me why the time is being added. What is the additional 10 seconds (10e3) being added to account for?

Reply:
Oh ok. That makes sense. Thank you.